language/vm/src/file_format.rs

```diff
@@ -404,19 +424,27 @@ pub struct FunctionDefinition {
 }
 
 impl FunctionDefinition {
-    /// Returns whether the FunctionDefinition is public.
+    /// Returns whether the FunctionDefinition can be called from a module other than the module
+    /// that defines it.
+    ///
+    /// TODO: following the definition above (which is inherited from the old public-private binary
+    /// model), the `protected` scope should also be considered as `is_public`. We need to refactor
+    /// the function name as well as its call stack.
     pub fn is_public(&self) -> bool {
-        self.is_public
+        matches!(&self.visibility, VisibilityScope::Public | VisibilityScope::Protected(_))
     }
```

Comment:
If you were actually going to land this bit by bit, it should return false on protected.

In actuality, I'd prefer to see a larger PR that refactored this into `pub fn visiblity(&self) -> &Visibility` 

This function is probably called in just one spot in the whole codebase (at least just a handful) so the refactor shouldn't be bad

Reply:
No, I am not going to land this :)

This is just for a proof-of-concept only and you are absolutely right that we need to make sure things are all well-defined and cleared up before making the changes.